src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/security/TokenVerifyingConfigFactory.kt

```diff
@@ -1,57 +1,55 @@
 package org.vaccineimpact.orderlyweb.security
 
+import org.pac4j.core.client.Client
+import org.pac4j.core.client.Clients
 import org.pac4j.core.config.Config
 import org.pac4j.core.config.ConfigFactory
+import org.pac4j.core.credentials.Credentials
 import org.pac4j.core.profile.CommonProfile
-import org.vaccineimpact.orderlyweb.models.permissions.PermissionSet
 import org.vaccineimpact.orderlyweb.db.TokenStore
+import org.vaccineimpact.orderlyweb.models.permissions.PermissionSet
 
 class TokenVerifyingConfigFactory(
         private val requiredPermissions: Set<PermissionRequirement>
 ) : ConfigFactory
 {
     companion object
     {
-        val headerClientWrapper = JWTHeaderClientWrapper(WebTokenHelper.instance.verifier)
-        val cookieClientWrapper = JWTCookieClientWrapper(WebTokenHelper.instance.verifier)
-        val parameterClientWrapper = JWTParameterClientWrapper(
-                WebTokenHelper.instance.verifier,
-                TokenStore.instance
-        )
+        val headerClient = JWTHeaderClient(WebTokenHelper.instance.verifier)
+        val cookieClient = JWTCookieClient(WebTokenHelper.instance.verifier)
+        val parameterClient = JWTParameterClient(WebTokenHelper.instance.verifier, TokenStore.instance)
+
     }
 
-    val clientWrappers = mutableListOf(headerClientWrapper, cookieClientWrapper)
+    val allClients = mutableListOf<OrderlyWebCredentialClient>(headerClient, cookieClient)
 
     override fun build(vararg parameters: Any?): Config
     {
-        clientWrappers.forEach {
-            it.client.addAuthorizationGenerator({ _, profile -> extractPermissionsFromToken(profile) })
-        }
-
-        return Config(clientWrappers.map { it.client }).apply {
+        @Suppress("UNCHECKED_CAST")
+        return Config(allClients as List<Client<Credentials, CommonProfile>>).apply {
```

Comment:
I remember now why we used the wrappers - you end up in type checking hell otherwise. Things work if we do this explicit cast here.